grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2913,7 +2914,7 @@ class StockMovementService {
                 stockMovement.isInApprovalState()
     }
 
-    def getApplicationTagLib() {
-        return grailsApplication.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
+    ApplicationTagLib getApplicationTagLib() {
+        return grailsApplication.mainContext.getBean(ApplicationTagLib)
```

Comment:
Unless you are sure that the getBean method takes a class as an argument, I would highly recommend passing a String argument. As you mentioned in the first comment on this PR, this was an issue with the package.

Reply:
Nevermind, it looks like getBean(Class) does exist.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/BeanFactory.html#getBean(java.lang.Class)

Reply:
yes, exactly. I've obviously checked if it works and it does. 
We also have this already existing in e.g. Product domain:
https://github.com/openboxes/openboxes/blob/742e5f54a8d2ba5a6cf75f2a99bf0c10d4fa6b0b/grails-app/domain/org/pih/warehouse/product/Product.groovy#L682-L684
![Screenshot from 2023-10-30 22-20-31](https://github.com/openboxes/openboxes/assets/93163821/6f8de476-e186-4e80-a164-98136918a2a2)

If you are fine with that one, please approve/merge it today if it's possible.